File: sdk/php/src/Resources/Hosting.php
<?php

namespace HiTechCloud\SDK\Resources;

class Hosting extends BaseResource
{
    /**
     * Get reinstall info
     */
    public function getReinstall(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/reinstall');
    }

    /**
     * Reinstall
     */
    public function reinstall(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/reinstall', $data);
    }

    /**
     * Get status
     */
    public function getStatus(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/status');
    }

    /**
     * Get bandwidth graphs
     */
    public function getBandwidthGraphs(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/bandwidth-graphs');
    }

    /**
     * Get bandwidth
     */
    public function getBandwidth(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/bandwidth');
    }

    /**
     * Reset password
     */
    public function resetPassword(int $serviceId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/reset');
    }

}

File: sdk/csharp/Resources/Hosting.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class Hosting : BaseResource
{
    public Hosting(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> ReinstallAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/reinstall", id));
    }

    public async Task<Dictionary<string, object?>> DoReinstallAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/reinstall", id), data);
    }

    public async Task<Dictionary<string, object?>> StatusAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/status", id));
    }

    public async Task<Dictionary<string, object?>> BandwidthGraphsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/bandwidth-graphs", id));
    }

    public async Task<Dictionary<string, object?>> BandwidthAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/bandwidth", id));
    }

    public async Task<Dictionary<string, object?>> ResetPasswordAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/reset", id), data);
    }
}
